tools/ibc-test/src/tests/ibc/sudt_erc20_transfer.rs

```diff
@@ -0,0 +1,577 @@
+            hasher.finalize(&mut ret);
+            ret
+        };
+        let ibc_sudt_transfer_type_script = packed::Script::new_builder()
+            .hash_type(ScriptHashType::Type.into())
+            .code_hash(TYPE_ID_CODE_HASH.pack())
+            .args(ibc_sudt_transfer_type_id_args[..].pack())
+            .build();
+        let ibc_sudt_transfer_output = packed::CellOutput::new_builder()
+            .lock(packed::Script::default())
+            .type_(Some(ibc_sudt_transfer_type_script.clone()).pack())
+            .build_exact_capacity(Capacity::bytes(ibc_sudt_transfer.len()).unwrap())
+            .unwrap();
+        let placeholder_witness = packed::WitnessArgs::new_builder()
+            .lock(Some(Bytes::from_static(&[0u8; 65])).pack())
+            .build();
+        let tx = TransactionBuilder::default()
+            .input(first_input)
+            .output(ibc_sudt_transfer_output)
+            .output_data(ibc_sudt_transfer.pack())
+            .witness(placeholder_witness.as_bytes().pack())
+            .build();
+        let tx = complete_tx(
+            &ckb_url,
+            &tx,
+            ckb_sender_lock_script.clone(),
+            ckb_sender_key,
+        )
+        .unwrap();
+        let tx_hash = send_transaction(&ckb_url, tx).unwrap();
+        log::info!("deployed ibc-sudt-transfer");
+        let sudt_transfer_contract_out_point = packed::OutPoint::new_builder()
+            .tx_hash(tx_hash.pack())
+            .index(0u32.pack())
+            .build();
+
+        // Create channel (ibc-sudt-transfer (st-lock) <-> ICS20TransferERC20)
+        log::info!("create channel");
+        let zero = &[0u8; 32];
+        let args = Args {
+            channel_contract_code_hash: zero,
+            channel_id: 0,
+            client_id: zero,
+            packet_contract_code_hash: zero,
+        }
+        .encode();
+        let ibc_sudt_transfer_lock = packed::Script::new_builder()
+            .hash_type(ScriptHashType::Type.into())
+            .code_hash(ibc_sudt_transfer_type_script.calc_script_hash())
+            .args(args.pack())
+            .build();
+        let port_a_binary = ibc_sudt_transfer_lock.calc_script_hash().unpack().0;
+        // CKB port id is hex(lock hash).
+        let port_a: PortId = hex::encode(port_a_binary).parse().unwrap();
+        // ICS20TransferERC20 is bound to port transfer in axon chain
+        // initialization.
+        let port_b: PortId = "transfer".parse().unwrap();
+        let bootstrap_options = BootstrapChannelOptions::default()
+            .order(Order::Unordered)
+            .version("ics20".parse().unwrap())
+            .bootstrap_with_random_ids(false);
+        let channels = bootstrap_channel_with_connection(
+            &chains.handle_a,
+            &chains.handle_b,
+            connection,
+            &DualTagged::new(port_a).as_ref(),
+            &DualTagged::new(port_b).as_ref(),
+            bootstrap_options,
+        )
+        .unwrap();
```

Comment:
why don't use provided test frame in Hermes to create channel connection ?

Reply:
The port id is only known after deployment.

Reply:
ok, if putting the deployment of `ibc-sudt-transfer` in the generation stage, the port id known as `lock_hash` can be written in the `consts.rs` file ?

Reply:
Yes, it'll be possible then.